tensorflow/python/lib/io/file_io.py

```diff
@@ -411,7 +424,11 @@ def get_matching_files_v2(pattern):
         for matching_filename in _pywrap_file_io.GetMatchingFiles(
             compat.as_bytes(pattern))
     ]
-  else:
+
+  if not isinstance(num_threads, int):
+    raise TypeError("`num_threads` must be a integer.")
+
+  if num_threads <= 1:
```

Comment:
Inside this if, we should just set `num_threads` to 1 / raise error if it is strictly below 1.

Then we can reuse the pool implementation, it should work with a single thread too, right?

Reply:
All right! I have modified as you said. 

The reason for handling the case of one separately is due to the overhead of the thread pool created, but this overhead indeed accounts for a small proportion in the current situation.